File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/myshop/then/ThenStage.java
package com.mycompany.myshop.testkit.dsl.port.myshop.then;

import com.mycompany.myshop.testkit.dsl.port.myshop.then.steps.ThenClock;
import com.mycompany.myshop.testkit.dsl.port.myshop.then.steps.ThenCountry;
import com.mycompany.myshop.testkit.dsl.port.myshop.then.steps.ThenProduct;

public interface ThenStage {
    ThenClock clock();

    ThenProduct product(String skuAlias);

    ThenCountry country(String countryAlias);
}

File: system-test/typescript/src/testkit/dsl/port/myShop/then/then-stage.ts
import type { ThenClock } from './steps/then-clock.js';
import type { ThenProduct } from './steps/then-product.js';
import type { ThenCountry } from './steps/then-country.js';

export interface ThenStage {
  clock(): ThenClock;
  product(skuAlias: string): ThenProduct;
  country(countryAlias: string): ThenCountry;
}
